File: app/src/main/java/com/jguerrerope/moviechallenge/di/ActivityModule.java
package com.jguerrerope.moviechallenge.di;

import com.jguerrerope.moviechallenge.ui.MovieDetailsActivity;
import com.jguerrerope.moviechallenge.ui.MovieListActivity;
import dagger.Module;
import dagger.android.ContributesAndroidInjector;

@Module
public abstract class ActivityModule {
    @ContributesAndroidInjector
    abstract MovieListActivity contributeMovieListActivity();

    @ContributesAndroidInjector
    abstract MovieDetailsActivity contributeMovieDetailsActivity();
}

File: app/src/main/java/com/jguerrerope/moviechallenge/di/ActivityModule.kt
package com.jguerrerope.moviechallenge.di

import com.jguerrerope.moviechallenge.ui.MovieListActivity
import com.jguerrerope.moviechallenge.ui.SearchMovieListActivity
import dagger.Module
import dagger.android.ContributesAndroidInjector

@Module
abstract class ActivityModule {
    @ContributesAndroidInjector
    internal abstract fun contributeMovieListActivity(): MovieListActivity

    @ContributesAndroidInjector
    internal abstract fun contributeSearchMovieListActivity(): SearchMovieListActivity
}

